### DISCUSSION TITLE
Help needed! Can't get paginate to work.

### DISCUSSION THREAD

**Discussion by Abdulhadi-Zakkour** [rebing/graphql-laravel#959] [Q&A]
Hi

I'm building a Laravel e-commerce app and want to paginate the queries. I opted to use the built-in pagination class as it is sufficient for my purposes.

I built the listings query as follows:

```
class ListingsQuery extends Query
{
    public function type(): Type
    {
        return GraphQL::paginate('ProductListingType', 'ProductListing');
        // return Type::listOf(GraphQL::type('ProductListing'));
    }

public function resolve($root, $args $context, ResolveInfo $resolveInfo, Closure $getSelectFields)
    {
           $listings =  ProductListing::with(['product'=>['taxonomies','productFiles','listing', 'shippingProfile']])
                ->where('status', ProductListing::STATUS_APPROVED)
                ->paginate($args['limit'], ['*'], 'page', $args['page']);
            return $listings;
    }
```

I used this query: 
```
query {
   Listings(keyword:"",limit:1 page:2){
    product{
      name shop{ name, slug}
    } price 
  }
}
```

When I use `return GraphQL::paginate('ProductListingType', 'ProductListing');` the result of the query looks like this:
```
{
    "data": {
        "Listings": {
            "product": null,
            "price": null
        }
    }
}
```
While the result of the query when the type is set to `return Type::listOf(GraphQL::type('ProductListing'));` is like this:
```
{
    "data": {
        "Listings": [
            {
                "product": {
                    "name": "product11",
                    "shop": {
                        "name": "Shop11",
                        "slug": "shop11"
                    }
                },
                "price": 48.37
            }
        ]
    }
}
```

I did not change pagination_type in the config file.
What am I doing wrong here?

**Comment by mfn** [comment]
Unfortunately from the given information it's not obvious where the problem is, certainly `Type::listOf()` isn't correct on the returned type in the context.

You have to debug this yourself, check what the query result itself looks like for a start.

Or create a new project and try to reproduce it and share it, then I can take a look.

**Reply by Abdulhadi-Zakkour** [reply]
I think my query was wrong. 
Instead of:
```
query {
   Listings(keyword:"",limit:1 page:2){
    product{
      name shop{ name, slug}
    } price 
  }
}
```

I should write:
```
query {
   data{
   Listings(keyword:"",limit:1 page:2){
    product{
      name shop{ name, slug}
    } price 
    }
  }
}
```

It works now. Thank you for helping.
